src/components/PostsList.tsx

```diff
@@ -3,52 +3,54 @@
 import classNames from 'classnames';
 import React from 'react';
 import { Post } from '../types/Post';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { setSelectedPost } from '../features/SelectedPost';
 
-type Props = {
-  posts: Post[];
-  selectedPostId?: number;
-  onPostSelected: (post: Post | null) => void;
-};
-
-export const PostsList: React.FC<Props> = ({
-  posts,
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+export const PostsList: React.FC = () => {
+  const { posts } = useAppSelector(state => state.posts);
+  const selectedPost = useAppSelector(state => state.selectedPost);
+  const selectedPostId = selectedPost.selectedPost?.id;
+  const dispatch = useAppDispatch();
+  const hendelSelect = (post: Post | null) => {
+    dispatch(
+      setSelectedPost(post?.id === (selectedPostId || null) ? null : post),
```

Comment:
```suggestion
      setSelectedPost(post.id === selectedPostId ? null : post),
```
Logical OR `||` in brackets seems redundant